### ISSUE TITLE
update cmake to properly allow includes to be inside of src

### ISSUE THREAD

**Issue by Rinzii** [cpp-gamedev/DogTales#20]
Currently, to include classes of DogTales you have to call to them like as follow:

```cpp
#include <src/player.hpp>
```

Which is not only inconvenient but also pretty dirty for use in the project and breaks many common conventions with how a project is structured in C++.

I suggest we separate the includes so they either work like as follows:

```cpp
// Either any of these style of includes
#include <player.hpp>
#include <dogtales/player.hpp> // This is my personal favorite
#include <game/player.hpp>
````

I'd also like to push a separation of includes and implementations where headers go into a `include` folder and implementation files go into a `src` folder, but this is of lesser importance than the above proposal.

Here is an example of how the project would look like in directories and how we would separate everything with cmake.
```
DogTales/
│
├── CMakeLists.txt
│
├── include/
│   └── dogtales/ (headers go here)
│       └── player.hpp
│
└── src/ (implementation goes here)
    └── player.cpp
```
The includes would look like as follows with the above project structure after properly being configured with cmake:

```cpp
#include <dogtales/player.hpp>
```

**Comment by karnkaul** [comment]
The `src/x.hpp` part was deliberate, because just `x.hpp` could clash with a system / third party header. Agree that `dog_tales/x.hpp` is ideal. FYI that won't need any changes in CMake, just to rename `src` to `dog_tales` and fixup includes in code.

**Comment by Rinzii** [comment]
> The `src/x.hpp` part was deliberate, because just `x.hpp` could clash with a system / third party header. Agree that `dog_tales/x.hpp` is ideal. FYI that won't need any changes in CMake, just to rename `src` to `dog_tales` and fixup includes in code.

Yeah I am aware if we just rename src then no changes are required in the cmake. :P

Also personally I'm not picky of having everything in a singular root file. Though I do still stand by that we should rename it away from src as its an extremely generic name. I'm fine with just naming it dog_tales though. ^^

**Comment by karnkaul** [comment]
Closed via #32
